src/library.py

```diff
@@ -700,7 +700,7 @@ def verify(self):
         ) > 11.11 + self.df["tol"]
 
     def check_bool(self) -> bool:
-        if len(self.result[self.result == True] > 0):
+        if self.result[0]:
```

Comment:
This is not what I meant, sorry if I confused you 😂. We shall have a chat for me to understand the intended binary flag for these new items. Base on my understanding of the description in library.json, you probably don't need to implement check_bool here, the default check_bool in superclass may be the right one to use.

Reply:
I am sorry that I didn't catch your point correctly Jerry :(. We should discuss this together and for now, I left this part as it was. However, I will make adjustments based on our discussion results later :).

Reply:
I was going to not include ```check_bool``` method but, it was not possible because ``check_bool``` method in the superclass is an abstract method so, ``check_bool``` method has to overwrite the class in the superclass. This is why I left it as was. Please let me know if this doesn't make sense to you!

Reply:
I got this fixed based on our discussion. Thank you for catching this up Jerry. I learned this new part in the code today.